### PR TITLE
SNOW-1056407 Cleanup metadata properties class. Migrate records tests to junit5

### PR DESCRIPTION
**Pull request by sfc-gh-lshcharbaty**
# Overview

SNOW-1056407
A preliminary PR for introducing a new connector timestamp property.

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [x] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [x] `No` - Simple refactoring and tests.
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/records/RecordService.java [hunk 1]
```diff
@@ -93,26 +96,9 @@ public class RecordService {
...
-  public RecordService(SnowflakeTelemetryService telemetryService) {
-    this.telemetryService = telemetryService;
-  }
-
   /** Record service with null telemetry Service, only use it for testing. */
   @VisibleForTesting
-  public RecordService() {
-    this(null);
-  }
+  public RecordService() {}
```
Comment (sfc-gh-lshcharbaty) [inline src/main/java/com/snowflake/kafka/connector/records/RecordService.java:101]: A new constructor accepting `Clock` is going to be introduced.
That's why this constructor is still marked as `@VisibleForTesting`

File: src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java [hunk 2]
```diff
@@ -12,64 +19,37 @@ public class SnowflakeMetadataConfig {
...
+        Optional.ofNullable(config.get(property))
+            .orElse(SnowflakeSinkConnectorConfig.SNOWFLAKE_METADATA_DEFAULT);
 
-    createtimeFlag = createtime;
-    topicFlag = topic;
-    offsetAndPartitionFlag = offsetAndPartition;
-    allFlag = all;
+    // Cannot use Boolean.parseBoolean in order to ensure backward compatibility.
+    // The equality check must be case sensitive.
+    return "true".equals(value);
```
Comment (sfc-gh-gjachimko) [inline src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java:44]: can't we break that contract? i mean - using case insensitive comparison wouldn't break old customers but would be a a nice improvement for new installations.

**Review by sfc-gh-gjachimko — APPROVED** [review]

File: src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java (hunk 2 above)
Comment (sfc-gh-lshcharbaty) [reply src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java:44]: I doubt anyone will use uppercase "TRUE" value to disable something.
I'd like this pr to introduce no changes, so I can change it in the following one.

Comment (sfc-gh-tzhang) [reply src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java:44]: Create a JIRA or add TODO to make sure this won't fall out of the cracks?

**Review by sfc-gh-tzhang — APPROVED** [review]
lgtm, thanks!

File: src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java (hunk 2 above)
Comment (sfc-gh-lshcharbaty) [reply src/main/java/com/snowflake/kafka/connector/records/SnowflakeMetadataConfig.java:44]: [It's done already](https://github.com/snowflakedb/snowflake-kafka-connector/pull/833/files#diff-cf910e10942ebdab56b8d5ab0021e53cf6ee64f7105482fcdca09a0c2ce41797R45) in one of the PRs following.

**Pull request merged**